manifests/0000_30_cluster-api_13_vap.yaml

```diff
@@ -0,0 +1,121 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: machine-api-machine-vap
+spec:
+  matchResources:
+    namespaceSelector:
+      matchLabels:
+        name: openshift-machine-api
+  paramRef:
+    namespace: openshift-cluster-api
+    parameterNotFoundAction: Deny
+    selector: {}
+  policyName: machine-api-machine-vap
+  validationActions: [Deny]
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: machine-api-machine-vap
+spec:
+  failurePolicy: Fail
+
+  paramKind:
+    apiVersion: cluster.x-k8s.io/v1beta1
+    kind: Machine
+
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["machine.openshift.io"]
+      apiVersions: ["v1beta1"]
+      operations:  ["UPDATE"]
+      resources:   ["machines"]
+
+  # Requests must satisfy every matchCondition to reach the validations
+  matchConditions:
+    - name: check-only-non-service-account-requests
+      expression: >-
+        !(request.userInfo.username in [
+            "system:serviceaccount:openshift-machine-api:machine-api-controllers",
+            "system:serviceaccount:openshift-cluster-api:machine-api-migration"
+        ])
+    - name: check-authoritativeAPI-clusterapi
+      expression: 'object.status.authoritativeAPI == "ClusterAPI"'
+    - name: check-param-match
+      expression: 'object.metadata.name == params.metadata.name'
```

Comment:
We could add a comment to say why we only consider these requests

Reply:
Not sure if 'these' is the service accounts, or the entire `matchConditions` block. Here's the explanation :) 

- Anything migration or mapi controller is excluded, we trust them.
- We only care (for this VAP) if the MAPI machine is non-authoritative.
- The final one is basically a filter, the VAP binding will provide every matching object of the correct kind to the VAP policy. This means we match the correct instance of our parameter and therefore evaluate the policy.